lib/services/role.go

```diff
@@ -2214,12 +2213,14 @@ func MarshalRole(role types.Role, opts ...MarshalOption) ([]byte, error) {
 // older instances. Makes a shallow copy if the conversion is necessary. The
 // passed in role will not be mutated.
 // DELETE IN 8.0.0
-func DowngradeRoleToV3(r *types.RoleV4) (*types.RoleV4, error) {
+func DowngradeRoleToV3(r *types.RoleV5) (*types.RoleV5, error) {
```

Comment:
This is marked for delete in 8.0, rather than update it can we just get rid of it?

Reply:
I've made some changes to role downgrading. I got rid of that and made a function to handle V5 -> V4 degradation properly.